server/server_test.go

```diff
@@ -2090,42 +2090,81 @@ func TestStartPositionFirstSequence(t *testing.T) {
 }
 
 func TestStartPositionSequenceStart(t *testing.T) {
-	s := RunServer(clusterName)
+	opts := GetDefaultOptions()
+	opts.ID = clusterName
+	opts.MaxMsgs = 10
+	s := RunServerWithOpts(opts, nil)
 	defer s.Shutdown()
 
 	sc := NewDefaultConnection(t)
 	defer sc.Close()
 
-	rch := make(chan bool)
-
-	cb := func(_ *stan.Msg) {
-		rch <- true
+	// Send more messages than max so that first is not 1
+	for i := 0; i < opts.MaxMsgs+10; i++ {
+		if err := sc.Publish("foo", []byte("hello")); err != nil {
+			t.Fatalf("Unexpected error on publish: %v", err)
+		}
+	}
+	// Check first/last
+	firstSeq, lastSeq := s.store.LookupChannel("foo").Msgs.FirstAndLastSequence()
+	if firstSeq != uint64(opts.MaxMsgs+1) {
+		t.Fatalf("Expected first sequence to be %v, got %v", uint64(opts.MaxMsgs+1), firstSeq)
+	}
+	if lastSeq != uint64(opts.MaxMsgs+10) {
+		t.Fatalf("Expected last sequence to be %v, got %v", uint64(opts.MaxMsgs+10), lastSeq)
 	}
 
-	// Start a subscriber with "Sequence" as start position.
-	// As of now, since there is no message, the call will fail.
-	sub, err := sc.Subscribe("foo", cb, stan.StartAtSequence(0))
-	if err == nil {
-		sub.Unsubscribe()
-		t.Fatal("Expected error on subscribe, got none")
+	rch := make(chan bool)
+	// Create subscriber with sequence below firstSeq, it should not fail
+	// and receive message with sequence == firstSeq
+	sub, err := sc.Subscribe("foo", func(m *stan.Msg) {
+		if m.Sequence == firstSeq {
+			rch <- true
+		}
+	}, stan.StartAtSequence(1), stan.SetManualAckMode(), stan.MaxInflight(1))
+	if err != nil {
+		t.Fatalf("Unexpected error on subscribe: %v", err)
 	}
+	// Make sure correct msg is received
+	if err := Wait(rch); err != nil {
+		t.Fatal("Did not get our message")
+	}
+	sub.Unsubscribe()
 
-	// Send a message now.
+	// Create subscriber with sequence higher than lastSeq, it should not
+	// fail, but not receive until we send a new message.
+	sub, err = sc.Subscribe("foo", func(m *stan.Msg) {
+		if m.Sequence == lastSeq+1 {
```

Comment:
You'll want to also check to see it's the first message received by the subscriber.